src/book.py

```diff
@@ -0,0 +1,96 @@
+import uuid
+import os
+import logging
+from datetime import datetime
+
+from flask import Blueprint, jsonify, request
+from cloudinary.uploader import upload
+from cloudinary.utils import cloudinary_url
+
+from src.models import Author, Book,  db
+from src.google import get_user_info
+from src.constants.http_status_codes import HTTP_201_CREATED, HTTP_400_BAD_REQUEST
+
+book_bp = Blueprint('book', __name__, url_prefix='/api')
+
+
+@book_bp.post('/books')
+def create_book():
+    file = request.files['image']
+    if not file:
+        return jsonify({
+            'error': "book image missing"
+        }), HTTP_400_BAD_REQUEST
+
+
+    # Author
+    book_author = request.form['author'].split(' ')
+    author_first_name = book_author[0]
+    author_last_name = ' '.join(book_author[1:])
+
+    # Book
+    title = request.form['title']
+    isbn = request.form['isbn']
+    language = request.form['language']
+    year_of_publication = request.form['year_of_publication']
+    category = request.form['category']
+    owner_id = request.form['owner_id']
+    
+    if not title:
+        return jsonify({
+            'error': "book's title missing"
+        }), HTTP_400_BAD_REQUEST
+
+    if not language:
+        return jsonify({
+            'error': "book's language missing"
+        }), HTTP_400_BAD_REQUEST
+        
+    if not owner_id:
+        return jsonify({
+            'error': "book's owner missing"
+        }), HTTP_400_BAD_REQUEST
+    
+    if not year_of_publication:
+        return jsonify({
+            'error': "book's year of publication missing"
+        }), HTTP_400_BAD_REQUEST
+    
+    if not (author_first_name and author_last_name):
+        return jsonify({
+            'error': "author's name missing"
+        }), HTTP_400_BAD_REQUEST
+        
+
+    # Upload image to cloudinary server
+    cloudinary_response = upload(file, folder="bookie-books")
+    
+    if not cloudinary_response:
+        return jsonify({
+            'error': "error uploading image"
+        }), HTTP_400_BAD_REQUEST
```

Comment:
should it be the user's fault if cloudinary has an issue? 🤔 
Maybe return a 500 or check what response cloudinary is giving and return the appropriate error code to the user?

Reply:
Cloudinary returns `400: Bad request.`

Reply:
So to clarify, every error response from Cloudinary is a 400 right?

Reply:
It now returns `500` when the upload failed.  (However should failure to upload the image stop a user from saving the book) we can use a default/placeholder image, in this context)

Cloudinary SDK returns `200` when the upload is successful and returns `501` otherwise. 

Initially i was referring to the error code from the API option at https://cloudinary.com/documentation/image_upload_api_reference#error_handling rather than the SDK upload approach, since thats what we used.

I tested various scenarios ways of making the upload fail deliberately, and they all return INTERNAL_SERVER_ERROR `500`.

Reply:
Regarding whether failure to upload an image should stop a user from saving a book is more of a business decision. What would you prefer?